File: src/main/java/org/example/gogoma/domain/marathon/dto/MarathonPreviewDto.java
package org.example.gogoma.domain.marathon.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.gogoma.domain.marathon.entity.MarathonType;
import org.example.gogoma.domain.marathon.enums.MarathonStatus;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MarathonPreviewDto {

    private int id;
    private String title;
    private LocalDateTime registrationStartDateTime;
    private LocalDateTime registrationEndDateTime;
    private LocalDateTime raceStartTime;
    private String location;
    private String city;
    private String region;
    private String district;
    private MarathonStatus marathonStatus;
    private String thumbnailImage;
    private String dDay;
    private List<String> courseTypeList;
    private List<MarathonType> marathonTypeList;
}

File: fe/app/src/main/java/com/example/gogoma/data/model/MarathonPreviewDto.kt
package com.example.gogoma.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class MarathonPreviewDto (
    val id: Int,
    val title: String,
    val registrationStartDateTime: String,
    val registrationEndDateTime: String,
    val raceStartTime: String,
    val location: String,
    val city: String,
    val region: String,
    val district: String,
    val marathonStatus: String,
    val thumbnailImage: String,
    val courseTypeList: List<String>,
    val marathonTypeList: List<MarathonType>,
    val dday: String
) : Parcelable